Robust.Client/GameStates/ClientGameStateManager.cs

```diff
@@ -629,16 +633,42 @@ public IEnumerable<EntityUid> ApplyGameState(GameState curState, GameState? next
             var xforms = _entities.GetEntityQuery<TransformComponent>();
             var xformSys = _entitySystemManager.GetEntitySystem<SharedTransformSystem>();
 
-            var toApply = new Dictionary<EntityUid, (bool EnteringPvs, GameTick LastApplied, EntityState? curState, EntityState? nextState)>();
-            var toCreate = new Dictionary<EntityUid, EntityState>();
             var enteringPvs = 0;
-
+            _toApply.Clear();
+            _toCreate.Clear();
             var curSpan = curState.EntityStates.Span;
+
+            // Create new entities
+            {
+                using var _ = _prof.Group("Create uninitialized entities");
```

Comment:
```suggestion

            // Create new entities
            // This is done BEFORE state application to ensure any new parents exist before existing children have their states applied, otherwise, we may have issues with entity transforms!
            {
                using var _ = _prof.Group("Create uninitialized entities");
```

Too lazy to checkout branch again.